### ISSUE TITLE
L1/L2 regularization possible? 

### ISSUE THREAD

**Issue by simonpun123** [unit8co/darts#2167]
Hello there. I am new to the darts package for time series forecasting. I found that it is very convenient to use and a lot of features are included. 

I didn’t find anything in the API that allows one to use L1/L2 regularization, which is commonly used in other machine learning applications. I just wonder how to train the models in darts (especially those deep learning based models) with L1/L2 regularization. Not sure if this makes sense for time-series forecasting. 

Thanks!

**Comment by madtoinou** [comment]
Hi @simonpun123,

For deep learning model, you can provide a loss function containing with L1/L2 regularization with `loss_fn` (just need to find a way to access the weights of the model). Also, depending on the optimizer, adding weight decay using the `optimizer_kwargs` argument can be equivalent to L2 regularization.

For regression model, you need to use the appropriate class such as `sklearn.linear_model.Lasso`.

**Comment by simonpun123** [comment]
Thank you very much!

**Comment by brianreinke95** [comment]
> Hi @simonpun123,
> 
> For deep learning model, you can provide a loss function containing with L1/L2 regularization with `loss_fn` (just need to find a way to access the weights of the model). Also, depending on the optimizer, adding weight decay using the `optimizer_kwargs` argument can be equivalent to L2 regularization.
> 
> For regression model, you need to use the appropriate class such as `sklearn.linear_model.Lasso`.

Hi!

We need to initializate some object with this model? How its done?

**Comment by madtoinou** [comment]
You can find an example in the [example notebook](https://unit8co.github.io/darts/examples/20-RegressionModel-examples.html#RegressionModel).
